Applications/ClientWebPortal/Source/webapp/src/utils/signal-manager/index.js

```diff
@@ -1,18 +1,31 @@
 import HashMap from 'hashmap';
 import uuid from 'uuid';
+import valueValidator from 'utils/value-validator';
 
 export default class {
   constructor() {
     this.listeners = new HashMap();
   }
 
   addListener(listener) {
+    if (listener == null) {
```

Comment:
The null check is redundant since valueValidator checks for null so this work is being duplicated.

Also instead of a RangeError, you want to throw a TypeError.

Also why is valueValidator an object?  Why not just have a function without it being bound to an entire class/object?

Furthermore why is this implementation using a uuid instead of just an int?  UUIDs are intended for decentralized systems where no consensus can be established between participants.  But in this case we have a singular manager, so this manager should just keep track of IDs by incrementing a variable.

Reply:
Right, null check is redundant in this case. Fixing the error to TypeError.

ValueValidator is there so that other classes can make use of it. In case they need to do parameter checks like the signal manager.